### PR TITLE
full features: Budget App Capstone Project

### PR DESCRIPTION
**Pull request by Cosmosha**
**The following requirements were implemented;**

**Splash screen**

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (categories page)**

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

**Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.

**"Add a new category" page**

**"Add a new transaction" page**

**The user fills out a form to create a new transaction with:**
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- 
**Testing requirements**
Create unit and integration tests for all the most important components of your RoR application.

**Technical requirements**
- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @Cosmosha  

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] Please add the old [PR link](https://github.com/Cosmosha/BudgetPal/pull/4) that you merged which had the first code review in this new PR.
- [ ] You didn't implement the changes asked by the previous reviewer. Please add the requested changes.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Comment by Cosmosha** [comment]
Hello @Omar-Muhamad, 
Thank you for making time to review my project, however, I would appreciate it if you could give me a tip on how I should add the merged link. Also please take time to read the comments to know the communication between me and the previous reviewer, this might give you an idea of the kind of changes I made, and why the old PR got merged after pushing a commit.
All things equal, thank you for the feedback.

**Comment by Cosmosha** [comment]
Please see the [link](https://github.com/Cosmosha/BudgetPal/pull/4) to the comments from previous reviewer from [previous PR](https://github.com/Cosmosha/BudgetPal/pull/4)

File: app/views/entities/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                <h4 class="card-title">Payment</h4>
+                <h4 class="card-subtitle mb-2">Total: $<%= @group.entities.sum(&:amount) %></h4>
+            </div>
+        </div>
+    </div>
+</div>
+
+<footer>
+    <div class="action">
+        <%= link_to "add new transaction", new_group_entity_path(group_id: @group.id) %>
```
Comment (Shedrack-Sunday) [inline app/views/entities/index.html.erb:53]: - [ ] As the previous reviewer mentioned, kindly use the right color for the button. You can see the color you used was turned off by me, and it is not the same as #5fb523` which you included but its not used 

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Cosmosha         👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Rails features ✔️
- Good Use of testing frameworks ✔️ 
    
    
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key points notes I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/entities/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    </div>
+
+    <div class="field tform mb-2">
+        <%= f.number_field :amount, placeholder:"Amount", autofocus: true, autocomplete: "name", class: "form-control form-control-lg" %>
+    </div>
+
+    <div class="actions">
+        <%= f.submit "Add Transaction", class: "btn btn-primary btn-lg btn-block splashBtn" %>
+    </div>
+<% end %>
```
Comment (Gambit142) [inline app/views/entities/new.html.erb:25]: - [OPTIONAL] The requirements state that the form page for adding a new _**transaction/record**_ should have a form input where the user can choose at least one existing category that the new record/transaction will belong to. Kindly implement this project requirement.

**Review by Gambit142 — APPROVED** [review]
Hi @Cosmosha 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Used Postgres as database for the application ✔️ 
- The database schema reflects the structure given ✔️ 
- Use of `devise` for authentication✔️ 
- Installed `rspec` gem for writing tests ✔️ 
- Wrote unit and integration tests ✔️

### Optional Suggestion

- Kindly consider the comment tagged _**[OPTIONAL]**_ under the review summary 👍 💻 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**